docs/rules/0148/uid-format.md

```diff
@@ -0,0 +1,79 @@
+---
+rule:
+  aip: 148
+  name: [core, '0148', uid-format]
+  summary: Annotate uid with UUID4 format.
+permalink: /148/uid-format
+redirect_from:
+  - /0148/uid-format
+---
+
+# `uid` format annotation
+
+This rule encourages the use of the `UUID4` format annotation on the `uid`
+field, as mandated in [AIP-148][].
+
+## Details
+
+This rule looks on for fields named `uid` and complains if it does not have the
+`(google.api.field_info).format = UUID4` annotation or has a format other than
+`UUID4`.
+
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message Book {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/Book"
+    pattern: "books/{book}"
+  };
+
+  string name = 1;
+  string uid = 2; // missing (google.api.field_info).format = UUID4
+}
+```
+
+**Correct** code for this rule:
+
+```proto
+// Correct.
+message Book {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/Book"
+    pattern: "books/{book}"
+  };
+
+  string name = 1;
```

Comment:
let's include the IDENTIFIER annotation on name fields in these examples, so that they're complete.

Reply:
Done!